src/pages/core-concepts/astro-components.md

```diff
@@ -3,151 +3,206 @@ layout: ~/layouts/Main.astro
 
-```jsx
+```astro
 ---
-// This area is TypeScript (and therefore JavaScript)!
-import MyComponent from './MyComponent.astro'
+// Import your components in your component script...
+import SomeComponent from './SomeComponent.astro';
 ---
-
-<html lang="en">
-  <head>
-    <meta charset="UTF-8">
-    <meta name="viewport" content="width=device-width, initial-scale=1.0">
-    <title>Document</title>
-  </head>
-  <body>
-    <MyComponent></MyComponent>
-  </body>
-</html>
+<!-- ... then use them in your HTML! -->
+<div>
+  <SomeComponent />
+</div>
+<!-- Astro components MUST start with an uppercase letter. 
+     This would NOT work as expected: <somecomponent /> -->
 ```
 
-### Data and Props
+📚 You can also import and use components from other frontend frameworks like React, Svelte, and Vue.  Read our guide on [Component Hydration](/core-concepts/component-hydration) to learn more.
+
+### Dynamic JSX Expressions
+
+Instead of inventing our own custom syntax for dynamic templating, we give you direct access to JavaScript values inside of your HTML, using something that feels just like [JSX](https://reactjs.org/docs/introducing-jsx.html).
 
-`.astro` components can define local variables inside of the Frontmatter script. These are automatically exposed to the content below.
+Astro components can define local variables inside of the Frontmatter script. Any script variables are then automatically available in the HTML template below.
 
-```jsx
+#### Dynamic Values
+
+```astro
 ---
-let name = 'world';
+const name = "Your name here";
 ---
+<div>
+  <h1>Hello {name}!</h1>
+</div>
+```
+#### Dynamic Attributes
 
-<main>
-    <h1>Hello {name}!</h1>
-</main>
+```astro
+---
+const name = "Your name here";
+---
+<div>
+  <div data-name={name}>Attribute expressions supported</div>
+  <div data-hint={`Use JS template strings to mix ${"variables"}.`}>So good!</div>
+</div>
 ```
 
-`.astro` components can also accept props when they are rendered. Public props are exposed on the `Astro.props` global.
+#### Dynamic HTML
 
-```jsx
+```astro
 ---
-const { greeting = 'Hello', name } = Astro.props;
+const items = ["Dog", "Cat", "Platipus"];
 ---
+<ul>
+  {items.map((item) => (
+    <li>{item}</li>
+  ))}
+</ul>
+```
+
+
+### Component Props
+
+An Astro component can define and accept props. Props are available on the `Astro.props` global in your frontmatter script.
 
-<main>
-    <h1>{greeting} {name}!</h1>
-</main>
+```astro
+---
+// Example: <SomeComponent greeting="(Optional) Hello" name="Required Name" />
+const { greeting = 'Hello', name } = Astro.props;
+---
+<div>
+    <h1>{greeting}, {name}!</h1>
+</div>
 ```
 
-To define the props which your component accepts, you may export a TypeScript interface or type named `Props`.
+You can define your props with TypeScript by exporting a `Props` type interface. Astro will automatically pick up any exported `Props` interface and give type warnings/errors for your project.
```

Comment:
We _will_ automatically pick it up, but don't right now. We should add a note to this effect and link to https://github.com/snowpackjs/astro/issues/595.

Reply:
made a note that this will be "in the future"